src/model.cpp

```diff
@@ -100,10 +100,28 @@ bool Model::doAddComponent(const ComponentPtr &component)
     return ComponentEntity::doAddComponent(component);
 }
 
-void Model::addUnits(const UnitsPtr &units)
+bool Model::addUnits(const UnitsPtr &units)
 {
-    mPimpl->mUnits.push_back(units);
-    units->setParent(shared_from_this());
+    if (units == nullptr) {
+        return false;
+    }
+
+    // Prevent adding multiple times to list.
+    if (hasUnits(units)) {
+        return false;
+    }
+
+    // Prevent adding to multiple models: create copy otherwise.
```

Comment:
why does adding a units multiple times create clones whereas components and variables result in re-parenting?

Reply:
Yep ... that's the million dollar question!  At present:
- adding a component to another place (model or component) moves the ownership.  
- adding a variable or units to other places or many times to the same place adds duplicates (which this PR fixes). 

This PR:
- duplicates units so that different models can make use of the same ones (seemed logical ...), and 
- moves variables (not sure about this ... see discussion in #620) in line with what happens to components.

Reply:
(you shouldn't be able to add the same units more than once to the same model though)

Reply:
yep - realised my England was confusing :)

But this inconsistency seems bad. I'd recommend having all the `addX` methods do the same re-parenting for now (status quo) and look to consistently change that behaviour in future if that is where we should go.

Reply:
Changed in 443088bc